src/sql/workbench/services/notebook/common/interface.ts

```diff
@@ -0,0 +1,27 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as azdata from 'azdata';
+import { URI } from 'vs/base/common/uri';
+import { Event } from 'vs/base/common/event';
+
+export interface INotebookInput {
+	defaultKernel: azdata.nb.IKernelSpec,
+	connectionProfile: azdata.IConnectionProfile,
+	isDirty(): boolean;
+	setDirty(boolean);
+	readonly notebookUri: URI;
+	updateModel(): void;
+	isDirty(): boolean;
+	readonly editorOpenedTimestamp: number;
+	readonly layoutChanged: Event<void>;
+}
+
+export function isINotebookInput(value: any): value is INotebookInput {
+	if (typeof value.defaultKernel === 'object' &&
```

Comment:
You might as well check the other properties as well.

For the class types (URI & Event) you can use instanceof, the rest should use typeof